### PR TITLE
Add some pretty printers for fetcher stuff to make Mock failures read…

### PR DESCRIPTION
**Pull request by AlCutter**
…able.

### CODE REVIEW

File: cpp/net/url_fetcher.cc [hunk 1]
```diff
@@ -276,5 +276,41 @@ ostream& operator<<(ostream& output, const UrlFetcher::Response& resp) {
...
+         << "headers: " << req.headers << endl
+         << "body: <<EOF" << endl
+         << req.body << "EOF" << endl;
+  return output;
+}
+
+ostream& operator<<(ostream& os, const UrlFetcher::Verb& verb) {
+  switch (verb) {
+    case UrlFetcher::Verb::GET:
+      os << "GET ";
```
Comment (pphaneuf) [inline cpp/net/url_fetcher.cc]: I don't like those trailing spaces... I can add them when needed, but I removing them would be annoying.

File: cpp/net/url_fetcher.cc [hunk 2]
```diff
@@ -276,5 +276,41 @@ ostream& operator<<(ostream& output, const UrlFetcher::Response& resp) {
...
+      break;
+  }
+  return os;
+}
+
+
+ostream& operator<<(ostream& os, const UrlFetcher::Headers& headers) {
+  os << "{";
+  for (const auto& h : headers) {
+    os << h.first << ": " << h.second;
```
Comment (pphaneuf) [inline cpp/net/url_fetcher.cc]: Nothing to separate each header?

**Comment by pphaneuf** [comment]
Review done, just minor things...

File: cpp/net/url_fetcher.cc (hunk 1 above)
Comment (AlCutter) [inline cpp/net/url_fetcher.cc]: removed

File: cpp/net/url_fetcher.cc (hunk 2 above)
Comment (AlCutter) [inline cpp/net/url_fetcher.cc]: added `endl`s

**Comment by AlCutter** [comment]
Squashed them in already since they're minor, ready to merge.

**Comment by pphaneuf** [comment]
Hmm, something happened in your squashing? Some other commits in there?

File: cpp/net/url_fetcher.cc [hunk 3]
```diff
@@ -276,5 +276,41 @@ ostream& operator<<(ostream& output, const UrlFetcher::Response& resp) {
...
 
+ostream& operator<<(ostream& output, const UrlFetcher::Request& req) {
+  output << "verb: " << req.verb << endl
+         << "url: " << req.url << endl
+         << "headers: " << req.headers << endl
+         << "body: <<EOF" << endl
+         << req.body << "EOF" << endl;
+  return output;
+}
+
```
Comment (pphaneuf) [inline cpp/net/url_fetcher.cc]: Only one line here (and before the preceding definition), but two later on? We should probably pick one for the file... (I vote for two!) :smiley:

**Pull request merged**